pkg/cmdline/cmdline.go

```diff
@@ -112,22 +112,20 @@ func progress(cli *cmdline, app, desc string, codes <-chan rpc.UpSummary_StatusC
 	start := time.Now()
 	done := make(chan string, 1)
 	go func() {
-		defer close(done)
-		for {
-			select {
-			case code := <-codes:
-				switch code {
-				case rpc.UpSummary_SUCCESS:
-					done <- fmt.Sprintf("%s: %s  (%.4fs)\n", cyan(app), passStr(desc), time.Since(start).Seconds())
-					return
-				case rpc.UpSummary_FAILURE:
-					done <- fmt.Sprintf("%s: %s  (%.4fs)\n", cyan(app), failStr(desc), time.Since(start).Seconds())
-					return
-				}
-			case <-cli.Done():
+		defer func() {
+			close(done)
+		}()
+		for code := range codes {
```

Comment:
perhaps, but it's structured this way to support increased verbosity, i.e. status code UpSummary_LOGGING, etc. in the future.